packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts

```diff
@@ -386,8 +387,8 @@ export class Pipeline extends PipelineBase {
       throw new Error('Only one of artifactBucket and crossRegionReplicationBuckets can be specified!');
     }
 
-    // @deprecated(v2): switch to default false
-    this.crossAccountKeys = props.crossAccountKeys ?? true;
+    this.crossAccountKeys = props.crossAccountKeys
+      ?? (FeatureFlags.of(this).isEnabled(cxapi.CODEPIPELINE_CROSS_ACCOUNT_KEYS_DEFAULT_VALUE_TO_FALSE) ? false : true);
```

Comment:
I dare to use the boolean ternary operator for readability.

Reply:
I would agree with your approach as other it will be a bit confusing. But a comment here would be even better to explain when feature flag is enabled, default set to true, other false.